src/Components/Aspire.Confluent.Kafka/ConfluentKafkaMetrics.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Diagnostics.Metrics;
+
+namespace Aspire.Confluent.Kafka;
+
+internal sealed class ConfluentKafkaMetrics
+{
+    private readonly IMeterFactory _meterFactory;
+    private readonly Meter _meter;
+    public Counter<long> Tx { get; }
+    public Counter<long> TxBytes { get; }
+    public Counter<long> TxMessages { get; }
+    public Counter<long> TxMessageBytes { get; }
+    public Counter<long> Rx { get; }
+    public Counter<long> RxBytes { get; }
+    public Counter<long> RxMessages { get; }
+    public Counter<long> RxMessageBytes { get; }
+
+    public ConcurrentQueue<Measurement<long>> ReplyQueueMeasurements { get; } = new ConcurrentQueue<Measurement<long>>();
+    public ConcurrentQueue<Measurement<long>> MessageCountMeasurements { get; } = new ConcurrentQueue<Measurement<long>>();
+    public ConcurrentQueue<Measurement<long>> MessageSizeMeasurements { get; } = new ConcurrentQueue<Measurement<long>>();
+
+    public ConfluentKafkaMetrics(IMeterFactory meterFactory)
+    {
+        _meterFactory = meterFactory;
+        _meter = _meterFactory.Create(ConfluentKafkaCommon.MeterName);
+
+        _meter.CreateObservableGauge(Counters.ReplyQueue, GetReplyQMeasurements);
+        _meter.CreateObservableGauge(Counters.MessageCount, GetMessageCountMeasurements);
+        _meter.CreateObservableGauge(Counters.MessageSize, GetMessageSizeMeasurements);
+
+        Tx = _meter.CreateCounter<long>(Counters.Tx);
+        TxBytes = _meter.CreateCounter<long>(Counters.TxBytes);
+        TxMessages = _meter.CreateCounter<long>(Counters.TxMessages);
+        TxMessageBytes = _meter.CreateCounter<long>(Counters.TxMessageBytes);
+        Rx = _meter.CreateCounter<long>(Counters.Rx);
+        RxBytes = _meter.CreateCounter<long>(Counters.RxBytes);
+        RxMessages = _meter.CreateCounter<long>(Counters.RxMessages);
+        RxMessageBytes = _meter.CreateCounter<long>(Counters.RxMessageBytes);
+    }
+
+    public static class Counters
+    {
+        private const string Prefix = "messaging.kafka.";
+        public const string ReplyQueue = Prefix + "replyq";
+        public const string MessageCount = Prefix + "msg_cnt";
+        public const string MessageSize = Prefix + "msg_size";
+        public const string Tx = Prefix + "tx";
+        public const string TxBytes = Prefix + "tx_bytes";
```

Comment:
> Total number of bytes transmitted to Kafka brokers

if we know sizes of individual requests, we should expose this metrics as a histogram (similar to [`http.client.request.body.size`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md#metric-httpclientrequestbodysize) and call it something like `messaging.kafka.request.body.size`

If we only know aggregated value then 
`messaging.kafka.network.io` (similarly to [`process.network.io`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/process-metrics.md) could be an option if we add `direction` attribute.

Also `messaging.kafka.network.tx`/`messaging.kafka.network.transmitted` and `messaging.kafka.network.rx`/`messaging.kafka.network.received` maybe even a better option to avoid adding a new attributes

Reply:
> Also `messaging.kafka.network.tx`/`messaging.kafka.network.transmitted` and `messaging.kafka.network.rx`/`messaging.kafka.network.received` maybe even a better option to avoid adding a new attributes

I opted for this.